common/test.c

```diff
@@ -501,8 +503,11 @@ p11_test_copy_setgid (const char *input)
 		return NULL;
 	}
 
-	path = strdup ("/tmp/test-setgid.XXXXXX");
-	assert (path != NULL);
+	tmpdir = getenv("TMPDIR");
+	if (tmpdir == NULL)
+		tmpdir = "/tmp";
+
+	assert(asprintf(&path, "%s/test-setgid.XXXXXX", tmpdir) >= 0);
```

Comment:
I meant to change it to:
```c
ret = asprintf (...);
assert (ret >= 0);
```
so that asprintf is still called even if the assert statement becomes no-op.

Anyway, thank you for the update, though I'm leaning toward #17.